packages/directory/tests/unit/components/dir-item-name-cell-test.js

```diff
@@ -0,0 +1,47 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+
+module('Unit | Component | dir-item-name-cell', function(hooks) {
+  setupTest(hooks);
+
+  test('links are constructed correctly', function(assert) {
+    assert.expect(4);
+
+    const subject = this.owner.factoryFor('component:dir-item-name-cell');
+
+    let unsavedReportComponent = subject.create({
+      value: {
+        serialize() {
+          return { data: { type: 'reports' } };
+        },
+        tempId: '12345',
+        id: null
+      }
+    });
+
+    assert.equal(
+      unsavedReportComponent.get('itemLink'),
+      'reports.report',
+      'Component builds the link based on the type of the model correctly'
+    );
+
+    assert.equal(unsavedReportComponent.get('itemId'), '12345', 'Component uses the tempId when the id is falsy');
+
+    let savedDashboardComponent = subject.create({
+      value: {
+        serialize() {
+          return { data: { type: 'dashboards' } };
+        },
+        id: 1
```

Comment:
Can we test with a model having both `tempId` and `id`?

Reply:
That shouldn't ever happen